extensions/identity-hub/src/main/java/org/eclipse/dataspaceconnector/identityhub/processor/CollectionsWriteProcessor.java

```diff
@@ -0,0 +1,51 @@
+/*
+ *  Copyright (c) 2022 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.identityhub.processor;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.eclipse.dataspaceconnector.identityhub.dtos.MessageResponseObject;
+import org.eclipse.dataspaceconnector.identityhub.dtos.MessageStatus;
+import org.eclipse.dataspaceconnector.identityhub.dtos.VerifiableCredential;
+import org.eclipse.dataspaceconnector.identityhub.store.HubObject;
+import org.eclipse.dataspaceconnector.identityhub.store.IdentityHubStore;
+
+import java.io.IOException;
+
+import static org.eclipse.dataspaceconnector.identityhub.dtos.MessageResponseObject.MESSAGE_ID_VALUE;
+
+/**
+ * Processor of "CollectionsWrite" messages, in order to write {@link HubObject}s into the {@link IdentityHubStore}.
+ */
+public class CollectionsWriteProcessor implements MessageProcessor {
+
+    private final IdentityHubStore identityHubStore;
+    private final ObjectMapper mapper;
+
+    public CollectionsWriteProcessor(IdentityHubStore identityHubStore) {
+        this.identityHubStore = identityHubStore;
+        this.mapper = new ObjectMapper();
```

Comment:
EDC TypeManager can be reused here https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/spi/core-spi/src/main/java/org/eclipse/dataspaceconnector/spi/types/TypeManager.java